### PR TITLE
Added Ajax request to get timelines from backend

### PR DESCRIPTION
**Pull request by HadarRose**
- tutorials used: https://stackoverflow.com/questions/42412006/serving-up-a-javascript-file-using-node-js

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
 <html>
     <body>
+       
         <p>Hello World!</p>
+        <div>
+            <button onclick="getTimeline()">Get Timeline</button>
+            <p id="timeline-para"></p>
+        </div>
+
+        <script type="text/javascript" src="src/js/basicRequests.js"></script>
```
Comment (kev-chan) [inline src/index.html]: put this in the <head> tag

File: src/js/basicRequests.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+function getTimeline() {
+    console.info('Requesting timeline from backend.');
+    let xhttp = new XMLHttpRequest();
+    xhttp.onreadystatechange = function(){ // more on readystates: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/readyState
+        if(this.readyState == 4 && this.status == 200){ // waits for request to be done (readyState 4) and successful (status 200)
```
Comment (kev-chan) [inline src/js/basicRequests.js]: are there any existing variables you can use for the state or status?

**Review by kev-chan — CHANGES_REQUESTED** [review]

File: src/js/basicRequests.js (hunk 2 above)
Comment (HadarRose) [reply src/js/basicRequests.js]: I can use XMLHttpRequest.DONE for the state (added, will be pushed with the other changes) but I can't find a variable for the status. Should I get it from a different source?

File: src/index.html (hunk 1 above)
Comment (HadarRose) [reply src/index.html]: I'm sorry, I'm not sure I understand which tag you are referring to, or what should be in it. Is the script supposed to be elsewhere?

Comment (kev-chan) [reply src/index.html]: <head>

Comment (kev-chan) [reply src/index.html]: head tag

Comment (kev-chan) [reply src/index.html]: read up about what goes in the head tag and please adjust as necessary

**Review by kev-chan — CHANGES_REQUESTED** [review]

**Comment by kev-chan** [comment]
are all scenarios handled?

**Comment by HadarRose** [comment]
@kev-chan added error handling, which I believe handles all scenarios that need handling.

**Comment by kev-chan** [comment]
same comment as before

**Pull request merged**